### PR TITLE
Centralize the way the sample apps yaml are being set to be deployed

### LINKED ISSUE #467: [FEATURE] Avoid the use of bookinfo as testing app or enable a way to set custom images
### Is this the right place to submit this?

- [X] This is not a question about how to use the sail-operator

### Feature Description

We can test the configuration in the e2e framework by using an alternative containers or adding the possibility to set custom images that can be set by the user to avoid pulling always upstream images to run the test

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by fjglira**
#### What type of PR is this?

- [X] Enhancement / New Feature
- [ ] Bug Fix
- [X] Refactor
- [ ] Optimization
- [ ] Test
- [ ] Documentation Update

#### What this PR does / why we need it:
This PR addresses these needs:
* Centralize the way that the pod yaml URL is being used across all the e2e tests to simplify the deployment of the sample application
* Stop using bookinfo app to validate sidecar injection
* Adding POD_SAMPLE_YAML_BASE_URL to enable a way to set custom yaml string that will allow user to use their own yaml with custom images

#### Which issue(s) this PR fixes:

Fixes #467

### CODE REVIEW

**Comment by fjglira** [comment]
I'll check tomorrow morning the failure in the suites: dual-stack and multicluster. For sure, I left some errors in the new way of setting the sample apps

File: tests/e2e/util/common/e2e_utils.go [hunk 1]
```diff
@@ -261,3 +263,46 @@ func InstallOperatorViaHelm(extraArgs ...string) error {
...
+		url = "samples/sleep/sleep.yaml"
+	case "helloworld":
+	case "sample":
+		url = "samples/helloworld/helloworld.yaml"
+	default:
+		return ""
+	}
+
+	// Get the base URL from the environment variable or use the default
+	baseURL := os.Getenv("POD_SAMPLE_YAML_BASE_URL")
```
Comment (MaxBab) [inline tests/e2e/util/common/e2e_utils.go:296]: Does it mean, that if I'm setting a custom "baseURL" based on the env var, the path for the yaml files should be identical to the mentioned in lines 279-290?

Comment (fjglira) [reply tests/e2e/util/common/e2e_utils.go:296]: Yes, I'll improve this to be able to set the full path

Comment (fjglira) [reply tests/e2e/util/common/e2e_utils.go:312]: @MaxBab take a look, I think the best way is to set each yaml individually. With this, wdyt?

Comment (MaxBab) [reply tests/e2e/util/common/e2e_utils.go:312]: @fjglira 

Cool!
I like this approach.

**Review by MaxBab — APPROVED** [review]

File: tests/e2e/util/common/e2e_utils.go [hunk 2]
```diff
@@ -261,3 +263,60 @@ func InstallOperatorViaHelm(extraArgs ...string) error {
 func UninstallOperator() error {
 	return helm.Uninstall("sail-operator", "--namespace", OperatorNamespace)
 }
+
+// GetYAMLPodURL returns the URL of the yaml file for the testing app.
+// args:
+// version: the version of the Istio to get the yaml file from.
+// appName: the name of the testing app. Example: helloworld, sleep, tcp-echo.
+func GetYAMLPodURL(version supportedversion.VersionInfo, appName string) string {
```
Comment (dgn) [inline tests/e2e/util/common/e2e_utils.go:271]: I know it was called that before, but `GetYAMLPodURL` is a strange name for this func. Isn't  `GetSampleYAML` more adequate? Not sure what the pod has to do with this?

File: tests/e2e/util/common/e2e_utils.go [hunk 3]
```diff
@@ -261,3 +263,60 @@ func InstallOperatorViaHelm(extraArgs ...string) error {
...
+	case "sleep":
+		url = "samples/sleep/sleep.yaml"
+	case "helloworld", "sample":
+		url = "samples/helloworld/helloworld.yaml"
+	default:
+		return ""
+	}
+
+	// Base URL logic
+	baseURL := os.Getenv("POD_SAMPLE_YAML_BASE_URL")
```
Comment (dgn) [inline tests/e2e/util/common/e2e_utils.go:312]: why the `POD` prefix?

**Review by dgn — COMMENTED** [review]
LGTM to me, good improvement! Only two nits

**Review by dgn — APPROVED** [review]

**Pull request merged**